query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -73,29 +77,49 @@ final case class PeriodicSamplesMapper(start: Long,
       case c: ChunkedRangeFunction[_] if valColType == ColumnType.HistogramColumn =>
         source.map { rv =>
           val histRow = if (hasMaxCol) new TransientHistMaxRow() else new TransientHistRow()
+          val rdrv = rv.asInstanceOf[RawDataRangeVector]
+          val pubInt =
+            if (functionId.exists(_.onCumulCounter) && stepFactorNotationUsed) rdrv.publishInterval.getOrElse(0L)
```

Comment:
I think, rather than adding another parameter to the PSM which already has a lot, we should encapsulate the extra padding by abstracting this as a new `WindowParam` AST/trait, which can be NoWindow, SimpleWindow, WindowWithStepPadding etc.

Reply:
I was aware of this, but chose to accept the additional param. It is hard for the planner to make this call so early. The padding depends on
* whether counter based function is used
* whether publish interval is even available
* step factor notation was used

Some of this information is available only at runtime.